app/js/docker/dockerService.js

```diff
@@ -66,6 +66,11 @@ function dockerService($http, actions, flux, alertService, configService) {
             }
         };
 
+        if (action === 'pullImage') {
+            // hack for pulling images
+            config.params = data;
+        }
+
```

Comment:
Yeah the second is a little more verbose but unifies the API down to one function (with some behind the scenes mapping) and removes the need for partial function application in dockerService (which in theory means less memory usage). I'd say leave it as is for now and we can refactor in another ticket.